### PR TITLE
Enable 2024.1.0

### PR DESCRIPTION
**Pull request by artanokhov**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -10,7 +10,6 @@ mkdir -p build
 
 cmake ${CMAKE_ARGS}                                                          \
     -DCMAKE_BUILD_TYPE=Release                                               \
-    -DENABLE_INTEL_GNA=OFF                                                   \
```
Comment (ilya-lavrenov) [inline recipe/build.sh:13]: let's enable NPU plugin

**Comment by artanokhov** [comment]
@conda-forge-admin, please rerender

File: recipe/bld.bat [hunk 2]
```diff
@@ -6,8 +6,8 @@ mkdir -p build
 cmake                                                                        ^
     -DCMAKE_INSTALL_PREFIX="%LIBRARY_PREFIX%"                                ^
     -DCMAKE_BUILD_TYPE=Release                                               ^
-    -DENABLE_INTEL_GNA=OFF                                                   ^
     -DENABLE_SYSTEM_TBB=ON                                                   ^
+    -DENABLE_INTEL_NPU=ON                                                    ^
```
Comment (ilya-lavrenov) [inline recipe/bld.bat:10]: it's automatically enabled. We need:
- add new output in conda forge recipe
- add new resources (level zero) which will be used by NPU (similarly to oneDNN for CPU / GPU)

**Comment by artanokhov** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 3]
```diff
@@ -359,6 +365,35 @@ outputs:
...
+      skip: True  # [not linux64 and not win64]
+    requirements:
+      build:
+        - {{ compiler('cxx') }}
+        - {{ compiler('c') }}
+        - {{ stdlib("c") }}
+        - cmake
+      host:
+        - tbb-devel
+        - pugixml
```
Comment (ilya-lavrenov) [inline recipe/meta.yaml:388]: TBB and pugixml are not required.

Comment (artanokhov) [reply recipe/meta.yaml:388]: Fixed

File: recipe/meta.yaml [hunk 4]
```diff
@@ -359,6 +365,33 @@ outputs:
         - test -f $PREFIX/lib/openvino-{{ version_postfix }}/libopenvino_intel_gpu_plugin.so  # [unix]
         - if not exist %LIBRARY_BIN%\\openvino-{{ version_postfix }}\\openvino_intel_gpu_plugin%SHLIB_EXT% exit 1  # [win]
 
+  - name: libopenvino-intel-npu-plugin
```
Comment (ilya-lavrenov) [inline recipe/meta.yaml:368]: let's add it to other places within this file where plugins are used

File: recipe/snippets/openvino_available_devices.c [hunk 5]
```diff
@@ -14,6 +14,7 @@ int main() {
     OV_CALL(ov_core_get_property(core, "CPU", "AVAILABLE_DEVICES", &ret));
 #if defined(OPENVINO_ARCH_X86_64) && !defined(__APPLE__)
     OV_CALL(ov_core_get_property(core, "GPU", "AVAILABLE_DEVICES", &ret));
+#    OV_CALL(ov_core_get_property(core, "NPU", "AVAILABLE_DEVICES", &ret));
```
Comment (ilya-lavrenov) [inline recipe/snippets/openvino_available_devices.c:17]:
```suggestion
    // OV_CALL(ov_core_get_property(core, "NPU", "AVAILABLE_DEVICES", &ret));
```

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**